tests/api/fuzzer.py

```diff
@@ -0,0 +1,319 @@
+                    #L.debug(f"Validate {k}/{repr(v)} against {kv_schema}")
+            else:
+                L.error("Don't understand what the additionalProperties means?")
+        else:
+            L.error(f"Can't understand schema at {schema_path}")
+            pretty_json(schema,max_depth=1)
+    elif "type" in schema  and  schema["type"]=="string":
+        if isinstance(resp, str):
+            L.debug(f"{repr(resp)} matches {repr(schema)} at {schema_path}")
+            return
+        L.error(f"{repr(resp)} does not match schema {repr(schema)} at {schema_path}")
+    elif "type" in schema  and  schema["type"]=="boolean":
+        if isinstance(resp, bool):
+            L.debug(f"{repr(resp)} matches {repr(schema)} at {schema_path}")
+            return
+        L.error(f"{repr(resp)} does not match schema {repr(schema)} at {schema_path}")
+    elif "type" in schema  and  schema["type"] in ("number","integer"):
+        if 'nullable' in schema  and  resp is None:
+            L.debug(f"{repr(resp)} matches {repr(schema)} at {schema_path} (because nullable!)")
+            return
+        if isinstance(resp, int):
+            L.debug(f"{repr(resp)} matches {repr(schema)} at {schema_path}")
+            return
+        L.error(f"{repr(resp)} does not match schema {repr(schema)} at {schema_path}")
+    else:
+        L.error(f"What to do with the schema? {type(resp)} at {schema_path}")
+        pretty_json(schema,max_depth=1)
+
+
+
+class GetPath(object):
+    def __init__(self, url, spec):
+        self.url    = url
+        self.req_params = {}
+        self.opt_params = {}
+        self.success    = None
+        self.failures   = {}
+        if 'parameters' in spec.keys():
+            for p in spec['parameters']:
+                name = p['name']
+                req  = p.get('required',False)
+                target = self.req_params if req else self.opt_params
+                target[name] = Param(name, p['in'], p['type'])
+                if 'default' in p:
+                    defs = p['default']
+                    if isinstance(defs,list):
+                        for d in defs:
+                            target[name].values.add(d)
+                    else:
+                        target[name].values.add(defs)
+                if 'enum' in p:
+                    for v in p['enum']:
+                        target[name].values.add(v)
+                if req and len(target[name].values)==0:
+                    print(f"FAIL: No default values in swagger for required parameter {name} in {self.url}")
+            for code,schema in spec['responses'].items():
+                if code=="200" and 'schema' in schema:
+                    self.success = schema['schema']
+                elif code=="200":
+                    L.error(f"200 reponse with no schema in {self.url}")
+                else:
+                    self.failures[code] = schema
```

Comment:
Success codes span the whole 2xx series. It’s not just `200` you should be considering here.

Reply:
This will be replaced with some code to pull any success codes that are defined out of the swagger. This hardcoded value is just a hack to get started.